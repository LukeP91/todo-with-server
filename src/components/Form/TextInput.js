import _ from 'lodash'
import { withStyles } from '@material-ui/core/styles'
import FormControl from '@material-ui/core/FormControl'
import FormHelperText from '@material-ui/core/FormHelperText'
import Input from '@material-ui/core/Input'
import InputLabel from '@material-ui/core/InputLabel'
import PropTypes from 'prop-types'
import React, { Component } from 'react'

const styles = {
  container: {
    marginBottom: '20px',
  },
}

class TextInput extends Component {
  renderInput = props => {
    const {
      field: { name, ...field },
      form: { errors },
      label,
      classes,
    } = props

    if (_.isEmpty(errors)) {
      return (
        <FormControl className={classes.container}>
          <InputLabel htmlFor="name-simple">{label}</InputLabel>
          <Input id="name-simple" {...field} name={name} />
        </FormControl>
      )
    } else {
      return (
        <FormControl aria-describedby="name-error-text" className={classes.container} error>
          <InputLabel htmlFor="name-error">{label}</InputLabel>
          <Input id="name-error" {...field} name={name} />
          <FormHelperText id="name-error-text">{errors[name]}</FormHelperText>
        </FormControl>
      )
    }
  }

  render() {
    return this.renderInput(this.props)
  }
}

TextInput.propTypes = {
  field: PropTypes.object,
  form: PropTypes.object,
  label: PropTypes.string,
}

export default withStyles(styles)(TextInput)
