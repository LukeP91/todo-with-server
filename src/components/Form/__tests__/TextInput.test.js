import React from 'react'
import renderer from 'react-test-renderer'

import TextInput from '../TextInput'

describe('TextInput', () => {
  it('renders correctly', () => {
    const field = { name: 'test', value: '' }
    const form = { error: {} }

    const tree = renderer.create(<TextInput field={field} form={form} label="test" />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  describe('when errors are passed', () => {
    it('renders TextInput with errors', () => {
      const field = { name: 'test', value: '' }
      const form = { error: { description: 'some error' } }

      const tree = renderer.create(<TextInput field={field} form={form} label="test" />).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
