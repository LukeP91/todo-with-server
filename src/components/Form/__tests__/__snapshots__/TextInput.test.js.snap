// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
<div
  className="MuiFormControl-root-2 TextInput-container-1"
>
  <label
    className="MuiFormLabel-root-11 MuiInputLabel-root-6 MuiInputLabel-formControl-7 MuiInputLabel-animated-10"
    data-shrink={false}
    htmlFor="name-simple"
  >
    test
  </label>
  <div
    className="MuiInput-root-18 MuiInput-formControl-19 MuiInput-underline-22"
  >
    <input
      aria-invalid={false}
      className="MuiInput-input-26"
      disabled={false}
      id="name-simple"
      name="test"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`TextInput when errors are passed renders TextInput with errors 1`] = `
<div
  className="MuiFormControl-root-2 TextInput-container-1"
>
  <label
    className="MuiFormLabel-root-11 MuiInputLabel-root-6 MuiInputLabel-formControl-7 MuiInputLabel-animated-10"
    data-shrink={false}
    htmlFor="name-simple"
  >
    test
  </label>
  <div
    className="MuiInput-root-18 MuiInput-formControl-19 MuiInput-underline-22"
  >
    <input
      aria-invalid={false}
      className="MuiInput-input-26"
      disabled={false}
      id="name-simple"
      name="test"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value=""
    />
  </div>
</div>
`;
